src/org/labkey/test/tests/list/ListTest.java

```diff
@@ -505,6 +507,52 @@ public void testNameTrimming()
         checker().withScreenshot().verifyEquals("Name not trimmed as expected", trimmedName, editList.getName());
     }
 
+    /* Issue 51572: Bug with creating a new list by uploading a csv file in "UTF-8 with BOM" format
+     */
+    @Test
+    public void testCreateListWithBOMFile() throws IOException
+    {
+        String listName = TestDataGenerator.randomDomainName("From BOM File", 4);
+        File bomFile = TestFileUtils.getSampleData("lists/TestUTF8_BOM.csv");
+
+        EditListDefinitionPage listEditPage = _listHelper.beginCreateList(getProjectName(), listName);
+        listEditPage.getFieldsPanel()
+                .setInferFieldFile(bomFile);
+
+        String keyName = "KeyValue";
+        listEditPage.selectKeyField(keyName);
+        listEditPage.clickSave();
+
+        List<FieldDefinition> fields = List.of(
+                new FieldDefinition(keyName, ColumnType.Integer),
+                new FieldDefinition("A", ColumnType.String),
+                new FieldDefinition("B", ColumnType.String),
+                new FieldDefinition("C", ColumnType.String)
+        );
+
+        // First line of the file are the column headers.
+        int rows = (int) Files.lines(bomFile.toPath()).count() - 1;
+        int columns = fields.size();
+
+        String[][] data = new String[rows][columns];
+
+        for (int i = 0; i < rows; i++) {
+            for (int j = 0; j < columns; j++) {
+                data[i][j] = switch (j)
+                {
+                    case 0 -> Integer.toString(i + 101);
+                    case 1 -> String.format("A%d", i + 2);
+                    case 2 -> String.format("B%d", i + 2);
+                    case 3 -> String.format("C%d", i + 2);
+                    default -> throw new IllegalStateException("Unexpected value: " + j);
+                };
+            }
+        }
```

Comment:
Why cant we have a constant expected value like `String [][] data = {
                {"101","A2","B2","C2"},
                {"102","A3","B3","C3"},
                {"103","A4","B4","C4"},
                {"104","A5","B5","C5"},
                {"105","A6","B6","C6"},
                {"106","A7","B7","C7"},
                {"107","A8","B8","C8"},
                {"108","A9","B9","C9"},
                {"109","A10","B10","C10"},
                {"110","A11","B11","C11"}
        };
    }`
?

Reply:
Initially I did, but that tied the test very tightly to the test data. This allows someone to change the number of rows in the test file without having to change the test code.

Reply:
In my opinion if someone is changing the file, they can change data here as well. Secondly data can be changed to anything based on future requirement without restricting to pattern here 

But i let you decide.